src/lisp/kernel/clos/streams.lisp

```diff
@@ -240,20 +245,23 @@ truename."))
   (:documentation
    "This is like CL:FILE-LENGTH, but for Gray streams."))
 
-(defgeneric stream-file-descriptor (stream &optional direction)
+(defgeneric stream-file-descriptor (stream direction)
   (:documentation
    "Return the file-descriptor underlaying STREAM, or NIL if not
-   available. DIRECTION must be either :INPUT, or :OUTPUT and is
-   supposed to discriminate in case STREAM is a bidirectional
-   stream. DIRECTION is supposed to default to :INPUT.
+   available. DIRECTION must be either :INPUT, :OUTPUT, :IO or
+   :PROBE."))
```

Comment:
Maybe restore the part of the comment explaining why there's a direction parameter?